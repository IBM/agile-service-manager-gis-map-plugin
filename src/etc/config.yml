#=======================================================================================================
#                      Topology mapper configuration properties
#=======================================================================================================

#=======================================================================================================
#                       Web app settings
# webServerPort:
#     The http port used by the mapping app to serve the application.
#     Default value: 3000

# webServerSecurePort:
#     The https port used by the mapping app to serve the application.
#     Default value: 3443
#=======================================================================================================
webServerPort: 3000
webServerSecurePort: 4443

#=======================================================================================================
#               Connection information for the ASM topology service REST API.
#
# asmHost:
#     The name or IP address of the server which hosts the 
#     ASM topology serviceREST API. 
#     Default value: localhost
#
# asmPort:
#     The port number on the host server which is used to access 
#     the ASM topology service REST API via HTTPS. The value must be 
#     an integer.
#     Default value: 443
#
# asmTenantId:
#     The unique tenant ID to use in ASM topology service REST 
#     requests in order to specify which tenant's data is 
#     required.
#     Default value: cfd95b7e-3bc7-4006-a4a8-a73a79c71255
#
# asmUsername:
#     The username to use for accessing the ASM backend service, 
#     if authentication is enabled.
#     Default value: asm
#
# asmPassword:
#     The password to use for accessing the ASM backend service,
#     if authentication is enabled.
#     Default value: asm
#
# asmRootPath:
#     The root URL path for the ASM topology service REST API. This 
#     must be set to match the corresponding ASM topology service 
#     configuration.
#     Default value: /1.0/
#
#=======================================================================================================
asmHost: localhost
asmPort: 443
asmTenantId: cfd95b7e-3bc7-4006-a4a8-a73a79c71255
asmUsername: asm
asmPassword: asm
asmRootPath: '/1.0/'
locationServiceHost: localhost
locationServicePort: 4443
#=======================================================================================================
#               Connection information for link back to ASM UI.
#
# asmUIURL:
#     The base URL to ASM Topology viewer page, with resource Id substitution.
#     The substitution will replace {RESOURCE_ID} with the select location marker id
#     Default value: https://localhost/aiops/cfd95b7e-3bc7-4006-a4a8-a73a79c71255/topology-viewer
#
#=======================================================================================================
asmUIURL: https://localhost/aiops/cfd95b7e-3bc7-4006-a4a8-a73a79c71255/topology-viewer



#=======================================================================================================
#                             Mapping - popup configuration
#
# popupIgnoreProperties:
#     Comma separated list of property names which are not to be show in the popup
#     details table. 
#     Default value: _executionTime,_modifiedAt,_observedAt,_startedAt,beginTime,changeTime,createTime,_compositeId,_compositeOfIds,_createdAt,_status
# 
# tooltipProperties:
#     Comma separated list of property names which are to be include in the mouse over
#     tooltip.
#     Default value: ''
#
#=======================================================================================================
popupIgnoreProperties: geolocation,_executionTime,_modifiedAt,_observedAt,_startedAt,beginTime,changeTime,createTime,_compositeId,_compositeOfIds,_createdAt,_status
tooltipProperties: ''

#=======================================================================================================
#                             Mapping - location marker configuration
#
# locationTypesConfig:
#     Location data Type array for config required to draw locations on the map. This can include zoom min and max level range between 0 and 15. 
#     If a group type is provided, vertexType: group must be set and the memberEntityTypes must be provided or locationStyle.
#     Default value: {}
#
# affectedRadiusPropertyNames:
#     Comma separated list of property names in ASM that represent affected radius
#     of a location. This value represents the raidus in metres.
#     Default value: ''
# 
# linkEdgeTypes:
#     Comma separated list of edgeTypes in ASM that are to be pulled and used to draw
#     links on the map. The links will be drawn if both ends of the link are represented
#     as location markers. The default value pulls all possible edgeTypes
#     Default value: *
# 
# linkColorPropertyNames:
#     Comma separated list of property names in ASM that are to be pulled and used to colour
#     the links that are drawn.
#     Default value: ''
# 
# returnComposites:
#     Specify if composite resources are using in the map location requests to ASM
#     Default value: true
#=======================================================================================================
locationTypesConfig:
  # - entityType: undergroundStation
  #   vertexType: resource
  # - entityType: interchangeStation
  #   vertexType: resource
  # - entityType: railStation
  #   vertexType: resource
  # - entityType: province
  #   vertexType: group
  #   minZoom: 0
  #   maxZoom: 2
  #   locationStyle: marker
  # - entityType: city
  #   vertexType: group
  #   minZoom: 3
  #   maxZoom: 7
  #   locationStyle: marker
  # - entityType: city
  #   vertexType: group
  #   minZoom: 8
  #   maxZoom: 15
  #   memberEntityTypes:
  #    - site
  - entityType: province
    source: locationService
    aggreationZoomLevel: 3
    tag: province%
    vertexType: resource
    minZoom: 0
    maxZoom: 2
    locationStyle: marker
  - entityType: city
    source: locationService
    aggreationZoomLevel: 8
    tag: city%
    vertexType: resource
    minZoom: 3
    maxZoom: 7
    locationStyle: marker
  - entityType: site
    source: locationService
    vertexType: resource
    minZoom: 8
    maxZoom: 15
affectedRadiusPropertyNames: ''
linkEdgeTypes: '*'
linkColorPropertyNames: ''
returnComposites: true

#=======================================================================================================
#                             Mapping - configuration
#
# updateRate:
#     Polling rate in miliseconds to pull and update the map data
#     Default value: 90000
#
# initialViewLocation:
#     Initial view center location when map loads
#     Default value: '51.505,-0.09'
#
# initialZoomLevel:
#     Initial view zoom level when map loads
#     Default value: 11
#
#=======================================================================================================
updateRate: 900000
initialViewLocation: '51.035535,-114.043873'
initialZoomLevel: 3