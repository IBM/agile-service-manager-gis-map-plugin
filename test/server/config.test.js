// Import the module under test
import * as config from '../../src/scripts/server/config';

const fs = require('fs');
jest.mock('fs');


describe('config', function() {
    test('readConfig: use defaults if YAML file does not exist', function() {
        // Mock the fs.readFileSync function so it throws an error
        fs.existsSync.mockImplementation(() => false);

        config.readConfig();
        let testSettings = config.settings();

        expect(testSettings).toMatchSnapshot();
    });

    test('readConfig: read config from YAML file', function() {
        // Mock the fs.readFileSync function so it does not actually try to read from a file
        // but instead returns a hardcoded string
        const yaml =
            'webServerPort: 1000' + '\n' +
            'proxyServiceHost: localhost' + '\n' +
            'proxyServicePort: 2000' + '\n' +
            'proxyServiceUsername: asm' + '\n' +
            'proxyServiceRootPath: /1.0/';
        fs.readFileSync.mockImplementation(() => yaml);
        fs.existsSync.mockImplementation(() => true);

        config.readConfig();
        let testSettings = config.settings();

        let expected = 1000;
        let actual = testSettings.webServerPort;
        expect(actual).toBe(expected);

        expect(testSettings).toMatchSnapshot();
    });

    test('readConfig: read config from environment variables', function() {
        process.env.UI_SERVER_PORT = 4000;
        process.env.PS_HOST = 'host.com';

        config.readConfig();
        let testSettings = config.settings();

        let expected = 4000;
        let actual = testSettings.webServerPort;
        expect(actual).toBe(expected);

        expected = 'host.com';
        actual = testSettings.proxyServiceHost;
        expect(actual).toBe(expected);


        // Clean up
        delete process.env.UI_SERVER_PORT;
        delete process.env.PS_HOST;
    });

    
});
