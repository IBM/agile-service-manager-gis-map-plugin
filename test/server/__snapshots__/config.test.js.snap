// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`config readConfig: read config from YAML file 1`] = `
Object {
  "affectedRadiusPropertyNames": "",
  "asmUIURL": "https://localhost:16311/ibm/console/inasm/topology.jsp?resourceId={RESOURCE_ID}",
  "boundaryEntityTypes": "",
  "boundaryPropertyNames": "",
  "latitudePropertyNames": "latitude",
  "linkColorPropertyNames": "",
  "linkEdgeTypes": "*",
  "locationEntityTypes": "location",
  "longitudePropertyNames": "longitude",
  "popupIgnoreProperties": "_executionTime,_modifiedAt,_observedAt,_startedAt,beginTime,changeTime,createTime,_compositeId,_compositeOfIds,_createdAt,_status",
  "proxyServiceHost": "localhost",
  "proxyServicePassword": "asm",
  "proxyServicePort": 2000,
  "proxyServiceRootPath": "/1.0/",
  "proxyServiceTenantId": "cfd95b7e-3bc7-4006-a4a8-a73a79c71255",
  "proxyServiceUsername": "asm",
  "tooltipProperties": "",
  "updateRate": 90000,
  "webServerPort": 1000,
  "webServerSecurePort": 3443,
}
`;

exports[`config readConfig: use defaults if YAML file does not exist 1`] = `
Object {
  "affectedRadiusPropertyNames": "",
  "asmUIURL": "https://localhost:16311/ibm/console/inasm/topology.jsp?resourceId={RESOURCE_ID}",
  "boundaryEntityTypes": "",
  "boundaryPropertyNames": "",
  "latitudePropertyNames": "latitude",
  "linkColorPropertyNames": "",
  "linkEdgeTypes": "*",
  "locationEntityTypes": "location",
  "longitudePropertyNames": "longitude",
  "popupIgnoreProperties": "_executionTime,_modifiedAt,_observedAt,_startedAt,beginTime,changeTime,createTime,_compositeId,_compositeOfIds,_createdAt,_status",
  "proxyServiceHost": "localhost",
  "proxyServicePassword": "asm",
  "proxyServicePort": 443,
  "proxyServiceRootPath": "/1.0/",
  "proxyServiceTenantId": "cfd95b7e-3bc7-4006-a4a8-a73a79c71255",
  "proxyServiceUsername": "asm",
  "tooltipProperties": "",
  "updateRate": 90000,
  "webServerPort": 3000,
  "webServerSecurePort": 3443,
}
`;
